plugins/modules/backup_vault.py

```diff
@@ -0,0 +1,340 @@
+        client.Client.exceptions.InvalidParameterValueException,
+        client.Client.exceptions.InvalidParameterValueException,
+        client.exceptions.MissingParameterValueException,
+        client.Client.exceptions.ServiceUnavailableException,
+        client.Client.exceptions.LimitExceededException,
+        client.Client.exceptions.AlreadyExistsException,
+        BotoCoreError,
+        ClientError,
+    ) as err:
+        module.fail_json_aws(err, msg="Failed to create Backup Vault")
+    return resp
+
+
+def tag_vault(module, client, tags, vault_arn, curr_tags=None, purge_tags=True):
+    """
+    Creates, updates, removes tags on a Backup Vault resource
+
+    module : AnsibleAWSModule object
+    client : boto3 client connection object
+    tags : Dict of tags converted from ansible_dict to boto3 list of dicts
+    vault_arn : The ARN of the Backup Vault to operate on
+    curr_tags : Dict of the current tags on resource, if any
+    dry_run : true/false to determine if changes will be made if needed
+    """
+
+    if tags is None:
+        return False
+
+    curr_tags = curr_tags or {}
+    tags_to_add, tags_to_remove = compare_aws_tags( curr_tags, tags, purge_tags=purge_tags)
+
+    if not tags_to_add and not tags_to_remove:
+        return False
+
+    if module.check_mode:
+        return True
+
+    if tags_to_remove:
+        remove = {k: curr_tags[k] for k in tags_to_remove}
+        tags_to_remove = ansible_dict_to_boto3_tag_list(remove)
+        try:
+            client.remove_tags(ResourceId=vault_arn, TagsList=tags_to_remove)
+        except (BotoCoreError, ClientError) as err:
+            module.fail_json_aws(err, msg="Failed to remove tags from the vault")
+
+    if tags_to_add:
+        tags_to_add = ansible_dict_to_boto3_tag_list(tags_to_add)
+        try:
+            client.add_tags(ResourceId=vault_arn, TagsList=tags_to_add)
+        except (BotoCoreError, ClientError) as err:
+            module.fail_json_aws(err, msg="Failed to add tags to Vault")
+
+    return True
+
+
+def get_tag_list(keys, tags):
+    """
+    Returns a list of dicts with tags to act on
+    keys : set of keys to get the values for
+    tags : the dict of tags to turn into a list
+    """
+    tag_list = []
+    for k in keys:
+        tag_list.append({"Key": k, "Value": tags[k]})
+
+    return tag_list
+
+
+def get_vault_facts(module, client, vault_name):
+    """
+    Describes existing vault in an account
+
+    module : AnsibleAWSModule object
+    client : boto3 client connection object
+    name : Name of the backup vault
+    """
+    # get Backup Vault info
+    try:
+        resp = client.describe_backup_vault(vault_name)
+    except (BotoCoreError, ClientError) as err:
+        module.fail_json_aws(err, msg="Failed to describe the Backup Vault")
+
+    # Now check to see if our vault exists and get status and tags
+    if resp:
+        try:
+            tags_list = client.list_tags(ResourceIdList=[resp["BackupVaultArn"]])
```

Comment:
I think we might be able to reuse https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/backup.py

Reply:
The rest of the tagging code (add / remove / ensure) should probably be moved into module_utils.backup